src/day_2/ddpg/solution/ddpg.py

```diff
@@ -0,0 +1,297 @@
+"""SAC implementation for Pendulum-v1 environment.
+
+Reference:
+    - https://github.com/higgsfield/RL-Adventure-2/blob/master/7.soft%20actor-critic.ipynb
+"""
+
+import random
+
+import gymnasium as gym
+import numpy as np
+import torch
+from torch import nn, optim
+
+USE_CUDA = torch.cuda.is_available()
+DEVICE = torch.device("cuda" if USE_CUDA else "cpu")
+
+
+class ReplayBuffer:
+    """Replay buffer for SAC."""
+
+    def __init__(self, capacity):
```

Comment:
```suggestion
    def __init__(self, capacity: int):
```

모든 코드에 type annotation을 표기하면 좋을 것 같은데 어떠신가요~?